yeastphenome/apps/datasets/views.py

```diff
@@ -357,45 +326,45 @@ def gene_datasets(request, gene_id):
     import numpy as np
 
     try:
-        gene = Gene.objects.get(id=gene_id)
+        gene = Gene.objects.get(pk=gene_id)
     except Gene.DoesNotExist:
         raise Http404
 
     links = [
         {"url": reverse("common:explorer"), "name": "Explore data"},
         {"url": reverse("genes:index"), "name": "Genes"},
         {
-            "url": reverse("genes:detail", args=[gene.systematic_name]),
-            "name": "%s/%s" % (gene.common_name, gene.systematic_name),
+            "url": reverse("genes:detail", args=[gene.id]),
+            "name": "%s" % gene,
         },
         {
-            "url": reverse("genes:datasets", args=[gene.systematic_name]),
+            "url": reverse("genes:datasets", args=[gene.id]),
             "name": "Phenotypic Scores",
         },
     ]
 
     # Derive datasets bins here
     datapoints = (
         Data.objects.filter(gene=gene)
-        .exclude(Q(value=None) | Q(value=Decimal("NaN")))
-        .order_by("-value")
-        .values_list("value")
+        .exclude(valuez__isnull=True)
+        .order_by("-valuez")
+        .values_list("valuez")
     )
+
     count, division = np.histogram([float(x[0]) for x in datapoints])
     counts = []
     for i, number in enumerate(count):
         counts.append({"count": number, "value": division[i + 1]})
 
-    context = get_gene_names_context()
-    context.update(
-        {
-            "gene": gene,
-            "links": links,
-            "active": "explorer",
-            "counts": counts,
-            "division": division,
-        }
-    )
+    # context = get_gene_names_context()
```

Comment:
Same question here!

Reply:
Yeah, same answer as above. Why do we need that context? As far I can tell, we have everything we need in the other lists and querysets.

Reply:
This is the same as above, I think I must have added it preparing to add the plots (and then didn't). Definitely okay to remove!